en/docs/learn/api-controller/advanced-topics/configuring-different-endpoint-types.md

```diff
@@ -0,0 +1,255 @@
+
+    -   **failoverEndpoints** field should be specified if the `endpointRoutingPolicy` is `failover`. This field contains the following.
+
+        - `production`: The primary production endpoint (not an array)
+        - `sandbox`: The primary sandbox endpoint (not an array)
+        - `productionFailovers`: An array which consists failover production endpoints
+        - `sandboxFailovers`: An array which consists failover sandbox endpoints
+
+    -   **awsLambdaEndpoints** field should be specified if the `endpointType` is `aws`. This field contains the following.
+
+        - `accessMethod`: The access method of awslambda endpoint. Mandatory to specify. (Values can be `role_supplied` or `stored`)
+        - `amznRegion`: Region where the endpoint is located. Can be chosen from [here](https://docs.aws.amazon.com/general/latest/gr/rande.html#regional-endpoints)
+        - `amznAccessKey`: Access Key for endpoint
+        - `amznSecretKey`: Access Secret for endpoint
+
+Let us discuss configuring each of the above endpoint type with examples, seperately, in the following sections.
+
+### HTTP/REST Endpoints
+
+This can be subdivided into three (3) main scenarios.
+
+#### 1. HTTP/REST Endpoint - Without load balancing or failover
+
+The following is an example `api_params.yaml` file for this scenario. (Note that you can ignore specifying the `endpointType` field if you want, because by default it will contain the value as `rest`.)
+
+!!! example
+    ```go   
+    environments:
+      - name: test
+        endpoints:
+            production:
+            sandbox:
+      - name: dev
+        endpointType: rest
+        endpoints:
+            production:
+                url: https://dev.wso2.com
+            sandbox:
+                url: https://dev.sandbox.wso2.com
+    ```
+
+#### 2. HTTP/REST Endpoint - With load balancing
+
+The following is an example `api_params.yaml` file for this scenario. (Note that, as mentioned in the above section, you can ignore specifying the `endpointType` field if you want, because by default it will contain the value as `rest`.)
+
+!!! example
+    ```go   
+    environments:
+      - name: test
+        endpoints:
+            production:
+            sandbox:
+      - name: dev
+        endpointType: rest
+        endpointRoutingPolicy: load_balanced
+        loadBalanceEndpoints:
+            production:
+                - url: https://dev1.wso2.com
+                - url: https://dev2.wso2.com
+            sandbox:
+                - url: https://dev1.sandbox.wso2.com
+                - url: https://dev2.sandbox.wso2.com
+            sessionManagement: transport
+            sessionTimeout: 5000
+    ```
+
+#### 3. HTTP/REST Endpoint - With failover
+
+The following is an example `api_params.yaml` file for this scenario. (Note that, as mentioned in the above sections, you can ignore specifying the `endpointType` field if you want, because by default it will contain the value as `rest`.)
```

Comment:
```suggestion
The following is an example `api_params.yaml` file for this scenario. (Note that, as mentioned in the above sections, you can ignore specifying the `endpointType` field if you want, because by default it contains the value as `rest`.)
```